vault/provider.go

```diff
@@ -764,6 +772,10 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 
 	client.SetMaxRetries(d.Get("max_retries").(int))
 
+	var lastStates []string
+	//Leverage Vault helpers for eventual consistency on login
+	client = client.WithResponseCallbacks(RecordMultipleStates(lastStates...))
```

Comment:
Remember that client.Clone won't preserve callbacks.  (Might not matter for your PR, just calling it out in case.)

Reply:
It looks like both `WithResponseCallbacks` and `WithRequestCallbacks` makes a shallow clone of the client, so I'm not sure that it will carry over things like headers set right above.

Reply:
@calvn and @ncabatoff I updated the `providerConfigure` method to return the client plus the underlying state variable referenced in the callbacks, so if we need to Clone, we can easily give the clone callbacks that make use of the same state reference. That should take care of the cloning headers aspect.

Syncing with Ben today at 1PM PDT to move forward with that